pkg/provisioner/resourcehandlers.go

```diff
@@ -164,14 +161,18 @@ func createConfigMap(obc *v1alpha1.ObjectBucketClaim, ep *v1alpha1.Endpoint, c k
 
 // Only the finalizer needs to be removed. The CM will be garbage collected since its
 // ownerReference refers to the parent OBC.
-func releaseConfigMap(cm *corev1.ConfigMap, c kubernetes.Interface) error {
+func releaseConfigMap(cm *corev1.ConfigMap, c kubernetes.Interface) (err error) {
 	if cm == nil {
 		return nil
 	}
 
-	logD.Info("ConfigMap is garbage collected after its finalizer is removed", "name", cm.Namespace+"/"+cm.Name)
+	cm, err = c.CoreV1().ConfigMaps(cm.Namespace).Get(cm.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	logD.Info("removing ConfigMaps's finalizer", "name", cm.Namespace+"/"+cm.Name)
 	removeFinalizer(cm)
```

Comment:
Possibly `removeFinalizer` should return the object?

Reply:
the object is passed by ref, so no need